server/src/main/kotlin/org/javacs/kt/config/Config.kt

```diff
@@ -0,0 +1,45 @@
+package org.javacs.kt.config
+
+import com.google.gson.*
+import org.eclipse.lsp4j.InitializeParams
+import org.javacs.kt.LOG
+import org.javacs.kt.storage.Storage
+import java.lang.reflect.Type
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+fun parseServerConfiguration(params: InitializeParams): ServerConfiguration? {
+    val gson = GsonBuilder().registerTypeHierarchyAdapter(Path::class.java, GsonPathConverter()).create()
+
+    var storage: Storage? = null
+
+    if (params.initializationOptions != null) {
+        val options = gson.fromJson(params.initializationOptions as JsonElement, InitializationOptions::class.java)
+
+        if (options.storagePath != null) {
+            if (Files.exists(options.storagePath) && Files.isDirectory(options.storagePath)) {
+                storage = Storage(options.storagePath)
+            }
+        }
+    }
+
+    return ServerConfiguration(storage)
+}
```

Comment:
Yeah, LSP4J uses Gson for some things.

I chose `kotlinx.serialization` because it's an official library, so I'd expect it to be quite stable and designed for Kotlin, as you said. I was also experimenting with some of the non stable serialization formats at first and we could potentially move to those later on with little code changes I think. But it does look a bit overkill right now, since we're using json. I can change it if you prefer.

Reply:
Using `kotlinx.serialization` sounds good (I'm always a bit careful about introducing new dependencies, but this seems useful enough to have it around in any case).